src/Libraries/Nop.Data/Extensions/AsyncIQueryableExtensions.cs

```diff
@@ -667,4 +667,15 @@ public static async Task<IPagedList<T>> ToPagedListAsync<T>(this IQueryable<T> s
 
         return new PagedList<T>(data, pageIndex, pageSize, count);
     }
+
+    /// <summary>
+    /// Returns an <see cref="IAsyncEnumerable{T}"/> that can be enumerated asynchronously.
+    /// </summary>
+    /// <typeparam name="T">Source sequence element type.</typeparam>
+    /// <param name="source">Source sequence.</param>
+    /// <returns>A query that can be enumerated asynchronously.</returns>
+    public static IAsyncEnumerable<T> ToAsyncEnumerable<T>(this IQueryable<T> source)
+    {
+        return AsyncExtensions.AsAsyncEnumerable(source);
+    }
```

Comment:
The existing calls to `ToAsyncEnumerable` were simply wrapping an `IEnumerable` into an `IAsyncEnumerable` wrapper and were not asynchronous.